slack-clone-compose-sample/src/main/kotlin/io/getstream/slack/compose/ui/channels/ChannelItem.kt

```diff
@@ -0,0 +1,197 @@
+            .clip(shape = RoundedCornerShape(6.dp))
+            .clickable(onClick = { onChannelClick(channel) })
+            .padding(horizontal = 8.dp)
+            .fillMaxSize(),
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        val user = channel.getOtherUser()!!
+
+        Box(
+            modifier = Modifier
+                .clip(ChatTheme.shapes.avatar)
+                .height(32.dp)
+                .width(30.dp),
+        ) {
+            UserAvatar(
+                modifier = Modifier
+                    .width(24.dp)
+                    .height(24.dp)
+                    .align(Alignment.CenterStart),
+                user = user
+            )
+            OnlineIndicator(
+                modifier = Modifier
+                    .align(Alignment.BottomEnd)
+                    .size(14.dp),
+                online = user.online
+            )
+        }
+
+        Spacer(Modifier.width(6.dp))
+        ChannelName(
+            channel = channel,
+            modifier = Modifier.weight(1f),
+        )
+
+        val unreadCount = channel.unreadCount ?: 0
+        if (unreadCount > 0) {
+            UnreadCountBadge(unreadCount)
+        }
+    }
+}
+
+/**
+ * Component that represents a distinct channel item.
+ *
+ * A distinct channel is a channel created without ID based on members.
+ *
+ * @param channel The channel to display.
+ * @param onChannelClick Handler for a single tap on an item.
+ * @param modifier Modifier for styling.
+ */
+@Composable
+fun DirectGroupChatItem(
+    channel: Channel,
+    onChannelClick: (Channel) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Row(
+        modifier = modifier
+            .clip(shape = RoundedCornerShape(6.dp))
+            .clickable(onClick = { onChannelClick(channel) })
+            .padding(horizontal = 8.dp)
+            .fillMaxSize(),
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        Box(
+            modifier = modifier
+                .clip(RoundedCornerShape(4.dp))
+                .size(24.dp)
+                .background(ChatTheme.colors.borders),
+        ) {
+            Text(
+                modifier = Modifier.align(Alignment.Center),
+                text = "" + (channel.memberCount - 1),
+                style = ChatTheme.typography.captionBold,
+                fontSize = 12.sp,
+                maxLines = 1,
+                overflow = TextOverflow.Ellipsis,
+                color = ChatTheme.colors.textHighEmphasis,
+            )
+        }
+        Spacer(Modifier.width(12.dp))
+        ChannelName(channel)
+    }
+}
+
+
+/**
+ * Component that represents a regular named channel.
+ *
+ * @param channel The channel to display.
+ * @param onChannelClick Handler for a single tap on an item.
+ * @param modifier Modifier for styling.
+ * */
+@Composable
+fun ChannelItem(
+    channel: Channel,
+    onChannelClick: (Channel) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Row(
```

Comment:
If we're seeing that a lot of these Row wrappers are being repeated, maybe we just need to customize the content?

Then provide something like a DefaultChannelContainer that does this setup, but just change the content that we pass in and make it `@Composable RowScope.() -> Unit`